Spokestack/RingBuffer.swift

```diff
@@ -81,10 +81,11 @@ final class RingBuffer <T> {
     func read() throws -> T {
         if self.isEmpty {
             throw RingBufferStateError.illegalState(message: "ring buffer is empty")
+        } else {
+            let value: T = self.data[self.rpos]
+            self.rpos = self.pos(self.rpos + 1)
+            return value
         }
-        let value: T = self.data[self.rpos]
```

Comment:
I'm wondering how you reached this line of code before you made this change if self.isEmpty returned true, and if swift exceptions really work like this, you have the same bug in your `write` function on the `self.isFull` check

Reply:
Yeah, that's...weird.

Reply:
I'm unable to replicate the magical thinking that led to this change.